std/getopt.d

```diff
@@ -1102,6 +1108,23 @@ unittest
     getopt(args, "fb1", &fb1, "fb2", &fb2, "tb1", &tb1);
     assert(fb1 && fb2 && !tb1);
 
+    // test keepEndOfOptions
+
+    args = (["program.name", "--foo", "nonoption", "--bar", "--", "--baz"]).dup;
```

Comment:
As an aside, I'm fairly new to Phobos development and must ask - what is the rationale behind the `(<array>).dup` pattern in these unit tests?